File: QTD2.Infrastructure/Model/ILA/CBT/CBTUpdateOptions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QTD2.Infrastructure.Model.ILA.CBTReleaseSetting
{
    public class CBTUpdateOptions
    {
        public bool CBTRequiredForCource { get; set; }
        public string CBTLearningContractInstructions { get; set; }
        public int DueDateAmount { get; set; }
        public Domain.Entities.Core.CBTAvailablity Availablity { get; set; }
        public bool ChangeDueDate { get; set; }
        public int EmpSettingsReleaseTypeId { get; set; }
    }
}

File: QTD.SPA/src/app/_DtoModels/ILA_CBT/CBTUpdateOptions.ts
import { CBT } from "./CBT";
import { CBTAvailablity } from "./CBTAvailablity";
import { CBTCreateOptions } from "./CBTCreateOptions";
import { CBTDueDateInterval } from "./CBTDueDateInterval";

export class CBTUpdateOptions extends CBTCreateOptions {
    changeDueDate: boolean;

    constructor(cBTRequiredForCource: boolean,
        cBTLearningContractInstructions: string,
        availablity: CBTAvailablity,
        dueDateAmount: number,
        empSettingsReleaseTypeId :string,
        changeDueDate: boolean) {
            super();
            this.cBTRequiredForCource = cBTRequiredForCource;
            this.cBTLearningContractInstructions = cBTLearningContractInstructions;
            this.availablity = availablity;
            this.dueDateAmount = dueDateAmount;
            this.empSettingsReleaseTypeId = empSettingsReleaseTypeId;
            this.changeDueDate = changeDueDate;
    }

    setChangeDueDate(dueDateAmount :number, dueDateTypeId:string){
      this.dueDateAmount = dueDateAmount;
      this.empSettingsReleaseTypeId = dueDateTypeId;
      this.changeDueDate = true;
    }
}
